vllm/worker/cpu_model_runner.py

```diff
@@ -361,3 +387,41 @@ def execute_model(
             sampling_metadata=sampling_metadata,
         )
         return output
+
+    @torch.inference_mode()
+    def profile_run(self) -> None:
+        model_config = self.model_config
+        vlm_config = self.vision_language_config
+
+        # Enable top-k sampling to reflect the accurate memory usage.
+        sampling_params = SamplingParams(top_p=0.99, top_k=self.vocab_size - 1)
+        max_num_batched_tokens = min(
+            self.scheduler_config.max_num_batched_tokens,
+            self.model_config.max_model_len,
+        )
+        max_num_seqs = self.scheduler_config.max_num_seqs
+
+        assert self.lora_config is None
+
+        # Run the model with the dummy inputs.
+        num_layers = self.model_config.get_num_layers(self.parallel_config)
+        kv_caches = [None] * num_layers
+
+        for seq_len in [1, max_num_seqs, max_num_batched_tokens]:
+            if vlm_config:
+                seq_data, dummy_multi_modal_data = MULTIMODAL_REGISTRY \
+                    .dummy_data_for_profiling(seq_len, model_config, vlm_config)
```

Comment:
nit: Let's avoid using `\` and use parentheses instead.
```suggestion
                seq_data, dummy_multi_modal_data = (MULTIMODAL_REGISTRY
                    .dummy_data_for_profiling(seq_len, model_config, vlm_config))
```